Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -880,27 +880,28 @@ function Set-AvsVMStoragePolicy {
     } 
 
     $ProtectedVMs = Get-ProtectedVMs 
-    if ($ProtectedVMs.Name.Contains($VMName)) {
-        Write-Error "Access denied to this VM." -ErrorAction Stop
-    }
-    $VM = Get-VM $VMName
-    if ($null -eq $VM) {
+    $VMList = Get-VM $VMName | Where-Object {-not $_.Name -in $ProtectedVMs.Name}
+
+    if ($null -eq $VMList) {
         Write-Error "Was not able to set the storage policy on the VM. Could not find VM with the name: $VMName" -ErrorAction Stop
     }
-    if (-not $(Get-SpbmEntityConfiguration $VM).StoragePolicy -in $VSANStoragePolicies) {
-        Write-Error "Modifying Storage policy on this VM is not supported" -ErrorAction Stop
-    }
-    Write-Host "Setting VM $VMName storage policy to $StoragePolicyName..."
-    try {
-        Set-VM -VM $VM -StoragePolicy $StoragePolicy -ErrorAction Stop -Confirm:$false
-    }
-    catch [VMware.VimAutomation.ViCore.Types.V1.ErrorHandling.InvalidVmConfig] {
-        Write-Error "The selected storage policy $($StoragePolicy.Name) is not compatible with this VM. You may need more hosts: $($PSItem.Exception.Message)" -ErrorAction Stop
-    }
-    catch {
-        Write-Error "Was not able to set the storage policy on the VM: $($PSItem.Exception.Message)" -ErrorAction Stop
+    foreach ($VM in $VMList) {
+        if (-not $(Get-SpbmEntityConfiguration $VM).StoragePolicy -in $VSANStoragePolicies) {
+            Write-Error "Modifying storage policy on $($VM.Name) is not supported" -ErrorAction Continue
+            continue
```

Comment:
I think we're good to go after this modification so that we can still change some partial set that matched the wildcard while ignoring the ones that don't. LGTM